File: frontend/src/types/activityLog.ts
export interface ActivityLog {
  id: number;
  user_id: number;
  user_name: string;
  action: string;
  entity_type: string;
  entity_id: number | null;
  description: string;
  created_at: string;
}

export interface ActivityLogFilters {
  entity_type?: string;
  user_id?: number;
  from_date?: string;
  to_date?: string;
  limit?: number;
}

File: backend/app/models/activity_log.py
from sqlalchemy import Column, DateTime, ForeignKey, Integer, String, Text
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func

from app.db.base import Base


class ActivityLog(Base):
    """
    SQLAlchemy model representing the activity_logs table.
    """

    __tablename__ = "activity_logs"

    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(
        Integer, ForeignKey("users.id", ondelete="CASCADE"), nullable=False
    )
    action = Column(String(255), nullable=False)
    entity_type = Column(String(100), nullable=False)
    entity_id = Column(Integer, nullable=True)
    description = Column(Text, nullable=False)
    created_at = Column(DateTime(timezone=True), server_default=func.now())

    # Relationships
    user = relationship("User")

    @property
    def user_name(self) -> str:
        """
        Helper property that returns the user's name as a string.
        """
        return self.user.name if self.user else ""
